scrape.py

```diff
@@ -261,26 +260,28 @@ def get_heartbeat(api_endpoint, access_token, response, lat, lng, lat_i, lng_i):
     return heartbeat
 
 
-def get_token(username, password):
-    return login_ptc(username, password)
-
-
 def login(lat_i, lng_i):
-    access_token = get_token(global_username, global_password)
-    if access_token is None:
-        raise Exception('[-] Wrong username/password')
+    while True:
+        access_token = login_ptc(global_username, global_password)
+        if access_token is None:
+            print('[-] Login failed, retrying')
+            time.sleep(1)
+            continue
```

Comment:
It will retry =X

The "wrong password" message before was too specific.
It's actually a generic "login failed" case, which can be (usually is?) for a retry-able reason.

The semantics aren't different from a supervised scraper imo, other than restarting less.